### DISCUSSION TITLE
New model: gpt4-x-alpaca-13b-native-4bit-128g !!

### DISCUSSION THREAD

**Discussion by OWKenobi** [oobabooga/text-generation-webui#727] [General]
Have a look at this:

https://huggingface.co/anon8231489123/gpt4-x-alpaca-13b-native-4bit-128g/tree/main

it is a llama trained on GTP4-outputs, heavily improving the output (it is claimed up to 90% of GTP-4 quality). It is already quantized, use the cuda-version, works out of the box with the parameters  --wbits 4 --groupsize 128

**Beware that this model needs around 23GB of VRAM, and you need to install the 4-bit-quantisation enhancement explained elsewhere.**

I'm not the author, I just got the link today, but I tried it and it works!

**Input**

     ### Instruction
    How to install oobabooga text-generation-webui                                    *edited. Was a mistake here.    

**Output of Alpaca-30b-int4 (two runs)** (not cherry picked)

        1. Go to https://github.com/oobabooga/text-generation-webui and click "Clone or Download" button.
        2. Click "Download ZIP".
        3. Unzip the downloaded file.
        4. Open the folder "text_generation_webui" and open index.html in your browser.
        5. Enter the desired input parameters (e.g., number of words, topic) and press "Generate Text".
        
        1. Go to https://github.com/oobabooga/text-generation-webui 
        2. Click "Clone or Download" 
        3. Select "Download ZIP" 
        4. Unzip the file into your desired directory 
        5. Open index.html with a browser 
        6. Enter the desired input parameters 
        7. Press "Generate Text"

 **Output of gpt4-x-alpaca-13b-native-4bit-128g (two runs)** (not cherry picked)

        1. Make sure you have a GitHub account and are signed in.
        2. Click on the following URL to open the repository page: https://github.com/oobabooga/text-generation-webui.
        3. On the right sidebar, click 'Clone this project'.
        4. A new window will appear with the option to name the repository; accept the default name or provide your own.
        5. Once the cloning process finishes, go back to the Github page and click on the repository name (e.g., text_generation_webui).
        6. You should now see a list of files and directories; locate the 'bin' directory and click on it.
        7. Right-click anywhere inside the 'bin' directory and select 'Open Folder'.
        9. Double-click on the file named 'webui' to launch it.
        10. The web interface should load, allowing you to customize settings and generate text.
        
        1. Make sure you have a GitHub account and are signed in.
        2. Clone the repository: https://github.com/oobabooga/text-generation-webui.
        3. Navigate to the root directory of the cloned repository (cd text-generation-webui).
        4. Install any necessary dependencies with npm (npm install).
        5. Open a browser and navigate to the project's public folder (http://localhost:8080/).
        6. The web interface should be available at http://localhost:8080/index.html.

For me the new model outperforms the way bigger old model both in fidelity and length. I am glad for some benchmarks from you! :D

**Comment by SenninOne** [comment]
I already tried the ggml, it was insane, at the same level that GPT3.5 at least, I used it to code and it did it! It even modify it to add features or make specific changes, I'll definitely try the HF one, just one question it says in the readme file that doesn't work with oobabooga,  did you tried it with textgen?

**Reply by OWKenobi** [reply]
it worked out of the box for me. You probably need the newest version of everything, like the ability to use groupsize etc.

Can you explain me how to run the ggml-file? In the readme it states it only works with llama.cpp.

**Comment by icopp** [comment]
I gave this a try. To me, it seems very inclined to either talk about requests instead of actually responding to them, or to give extremely terse responses of no more than three or four sentences no matter how much coaxing you give it. Maybe there's something I'm missing in the prompting for anything moderately complex.

**Reply by enn-nafnlaus** [reply]
My experience as well.  Very disappointing.

**Reply by patrickmros** [reply]
Has someone found out what's the matter with it? I read everywhere how good this model is but I encountered the same problems as you.

**Reply by BarfingLemurs** [reply]
update, the issue is solved fpr me as of this commit 843f67222709c3523339fcab968d7926ab5c93a8
It had previously output comma, then something useless.

**Comment by BetaDoggo** [comment]
It's the best instruct model I've used so far. It doesn't really do chain responses like gpt4all but it's far more consistent and it never says no. Settings I've found work well: 
`temp = 0.800000, top_k = 40, top_p = 0.950000, repeat_penalty = 1.100000`

**Reply by St33lMouse** [reply]
Never says no is the key. If you don't mind getting requests turned down, receiving a moral lecture, getting politically adjusted "facts", getting your chats coldly examined by an unfriendly AI or getting reported to the authorities you should just save yourself a whole lot of hassle and sign up for ChatGPT.  But if you want an AI that will do as it is told, and will (eventually) have a personality and behavior tailored to your own choices, you'll need to go local.

**Comment by ye7iaserag** [comment]
is it better for chat too? or just instruct?

**Reply by vractal** [reply]
seems to work well for chat too (im using the same instruction template, just continuing the conversation as a chat)

**Comment by Jimmy5-UI** [comment]
HOW TO RUN THIS

**Comment by Crocky-101** [comment]
Hey, just trying to experiment in this space, and could use some help. I've mostly just used the basic llama-7b successfully so far. You mention this working out of the box, but I get a simple "Could not find the quantized model in .pt or .safetensors format" error when I try to load the model by its name, or select it from parameters. What other step might there be? (I'm on windows using conda)

**Reply by enn-nafnlaus** [reply]
It's kind of a buggy way it picks filenames.  TL/DR to see what files it's trying to open, run "strace python ..." rather than just "python ..."

**Reply by Crocky-101** [reply]
Good recommendation! I did try, but found it only reads the filenames of packages, and doesn't reference the models folder, even when I successfully start a model, like 4bit llama:
[llama-13b-4bit start output.txt](https://github.com/oobabooga/text-generation-webui/files/11169986/llama-13b-4bit.start.output.txt)
(though to clarify I believe I have an issue with torch as well, this model crashes if I start generation)

**Comment by OWKenobi** [comment]
Hello Crocky,

you have to respect naming conventions. Your folder where you unpack everything has to have the exact same name as the model, or it won't be able to find it.

**Reply by Crocky-101** [reply]
I have used the same name on the .pt file as on the folder, and use that name on the command (tried the same with the -cuda variant to be sure). But thank you for pointing that out! :)

**Comment by alpkabac** [comment]
Can I run this in colab?

**Comment by daveckw** [comment]
I face issue below. Any idea what is going on?

Loading gpt4-x-alpaca-13b-native-4bit-128g...
Loading model ...
Done.
Traceback (most recent call last):
  File "E:\Stable Diffussion\oobabooga-windows\text-generation-webui\server.py", line 302, in <module>
    shared.model, shared.tokenizer = load_model(shared.model_name)
  File "E:\Stable Diffussion\oobabooga-windows\text-generation-webui\modules\models.py", line 176, in load_model
    tokenizer = LlamaTokenizer.from_pretrained(Path(f"{shared.args.model_dir}/{shared.model_name}/"), clean_up_tokenization_spaces=True)
  File "E:\Stable Diffussion\oobabooga-windows\installer_files\env\lib\site-packages\transformers\tokenization_utils_base.py", line 1811, in from_pretrained
    return cls._from_pretrained(
  File "E:\Stable Diffussion\oobabooga-windows\installer_files\env\lib\site-packages\transformers\tokenization_utils_base.py", line 1965, in _from_pretrained
    tokenizer = cls(*init_inputs, **init_kwargs)
  File "E:\Stable Diffussion\oobabooga-windows\installer_files\env\lib\site-packages\transformers\models\llama\tokenization_llama.py", line 96, in __init__
    self.sp_model.Load(vocab_file)
  File "E:\Stable Diffussion\oobabooga-windows\installer_files\env\lib\site-packages\sentencepiece\__init__.py", line 905, in Load
    return self.LoadFromFile(model_file)
  File "E:\Stable Diffussion\oobabooga-windows\installer_files\env\lib\site-packages\sentencepiece\__init__.py", line 310, in LoadFromFile
    return _sentencepiece.SentencePieceProc
![2023-04-09_182224](https://user-images.githubusercontent.com/34201653/230767374-e891f7a9-1a3f-4885-bc9e-1934c22bf065.png)
essor_LoadFromFile(self, arg)
RuntimeError: Internal: D:\a\sentencepiece\sentencepiece\src\sentencepiece_processor.cc(1102) [model_proto->ParseFromArray(serialized.data(), serialized.size())]

**Reply by kadattack** [reply]
Same, anyone know why this is happening?  It comes from `tokenizer = LlamaTokenizer.from_pretrained(Path(f"{shared.args.model_dir}/{shared.model_name}/"), clean_up_tokenization_spaces=True)`
Search on the net says that there's an issue with the tokenizer and nothing more. I tried checking if the path of the model folder was ok and it was.

**Reply by ExtinctionEvent** [reply]
Could this be the issue with the renaming of the LLaMATokenizer to LlamaTokenizer?

**Comment by MancV21** [comment]
Get an out of memory error running this. Have 16GB RAM and 12GB of VRAM. Can you tell me what spec you all have that have it running? Also, what command line do you have in the start.bat?

**Reply by OWKenobi** [reply]
Hi, indeed, you need about 23GB of VRAM. With 12GB, you can only run the 7b-models, as far as I know.

**Reply by Crocky-101** [reply]
Running these 4-bit models helps a lot with this. I can run normal LLaMA 13B 4-bit on 10GB VRAM / 32GB CPU RAM. Does this model also support using the —pre_layer flag? By only running 12-16 layers on GPU, I can even run the LLaMA 30B 4-bit, just very slowly

**Comment by ThioJoe** [comment]
I get answers like this no matter what I ask it lol:
<img width="636" alt="image" src="https://user-images.githubusercontent.com/12518330/231020519-52febf9b-9c90-4855-a61a-54544b451dab.png">

Tried both non-cuda and cuda. Any ideas?

**Reply by jllllll** [reply]
> I get answers like this no matter what I ask it lol: <img alt="image" width="636" src="https://user-images.githubusercontent.com/12518330/231020519-52febf9b-9c90-4855-a61a-54544b451dab.png">
> 
> Tried both non-cuda and cuda. Any ideas?

https://github.com/oobabooga/text-generation-webui/issues/931#issuecomment-1501259027

**Reply by OWKenobi** [reply]
Try to look at nvidia-smi if you run out of memory while generating. I had it once tell garbage when I activated beam search, which was too much to handle. Or, you have a partial download on the model file. Double-check if it downloaded completely.

**Reply by ccs96307** [reply]
Try to keep only the following files in your `models/gpt4-x-alpaca-13b-native-4bit-128g/`:
- config.json
- generation_config.json
- gpt-x-alpaca-13b-native-4bit-128g-cuda.pt
- special_tokens_map.json
- tokenizer.model
- tokenizer_config.json

and remove others.

**Reply by ExtinctionEvent** [reply]
works! thanks

**Reply by ThioJoe** [reply]
Yea for me the solution was simply to remove the non-cuda model file from the folder.

**Comment by TvoEx** [comment]
This model works with pre_layer, so I can even run it with merely 8Gb of VRAM, albeit slowly (0.5 tokens/s) using --pre_layer 20

**Reply by jllllll** [reply]
You should be able to run it with more than 20 layers with 8gb VRAM. The model has 40 layers and each layer requires ~0.222gb of memory. With ~1gb of VRAM spare for inference, that leaves 31 layers that you can fit on GPU.

**Reply by TvoEx** [reply]
I guess I was just using a bit of VRAM elsewhere when I tried that.

**Comment by Toron458** [comment]
with 2070TI Super 8 Gig VRAM:
call python server.py --auto-devices --chat --wbits 4 --groupsize 128 --pre_layer 17 --rwkv-cuda-on --model gpt4-x-alpaca-13b-native-4bit-128g-cuda
Most of the time it also works with --pre_layer 19 but i had times where it was not enough. It's really slow with around 0.5 token/s but it is stable. Anything higher then 19 did end up with OOM msg when you try chatting for longer answers.

**Comment by NetAndif** [comment]
I have 24 GB VRAM and the model "gpt-x-alpaca-13b-native-4bit-128g-cuda" has been loaded, but the interface shows the following error:

```
Traceback (most recent call last):
File “D:\oobabooga-windows\text-generation-webui\[server.py](http://server.py/)”, line 96, in load_model_wrapper
shared.model, shared.tokenizer = load_model(shared.model_name)
File “D:\oobabooga-windows\text-generation-webui\modules\[models.py](http://models.py/)”, line 196, in load_model
tokenizer = LlamaTokenizer.from_pretrained(Path(f"{shared.args.model_dir}/{shared.model_name}/“), clean_up_tokenization_spaces=True)
File “D:\oobabooga-windows\installer_files\env\lib\site-packages\transformers\tokenization_utils_base.py”, line 1811, in from_pretrained
return cls.from_pretrained(
File “D:\oobabooga-windows\installer_files\env\lib\site-packages\transformers\tokenization_utils_base.py”, line 1965, in from_pretrained
tokenizer = cls(*init_inputs, **init_kwargs)
File “D:\oobabooga-windows\installer_files\env\lib\site-packages\transformers\models\llama\tokenization_llama.py”, line 96, in init
self.sp_model.Load(vocab_file)
File "D:\oobabooga-windows\installer_files\env\lib\site-packages\sentencepiece_init.py", line 905, in Load
return self.LoadFromFile(model_file)
File "D:\oobabooga-windows\installer_files\env\lib\site-packages\sentencepiece_init.py”, line 310, in LoadFromFile
return _sentencepiece.SentencePieceProcessor_LoadFromFile(self, arg)
RuntimeError: Internal: D:\a\sentencepiece\sentencepiece\src\sentencepiece_processor.cc(1102) [model_proto->ParseFromArray(serialized.data(), serialized.size())]
```

**Comment by ozzafar** [comment]
is it possible to run in in Google Colab?
can anyone share please a notebook?

**Reply by Bortus-AI** [reply]
You can use this 
https://github.com/camenduru/text-generation-webui-colab

**Comment by echo-saurav** [comment]
is there any way to run it without gpu? like llama.cpp?
